### PR TITLE
v12 hgg subtyping (10/N) 

### PR DESCRIPTION
**Pull request by ewafula**
#### What scientific question is your analysis addressing?
Update HGG subtyping for v12  data release.

#### What was your approach?
- update all outmoded gene symbols 
- update most of the 10 module scripts
- update logic for IHG subtyping to include  methylation 
- updated HGG/DMG/IDH to include methylation 
- regenerated new results with v12 pre-release datasets
- updated README

#### What GitHub issue does your pull request address?
- https://github.com/PediatricOpenTargets/ticket-tracker/issues/487

#### Which areas should receive a particularly close look?
- Review all updated module scripts to check if the logic makes sense. Specifically, the subtyping script 07 is revamped to include methylation classification in the HGG subtyping. 
- compare with the previous updated IHG subtyping and which was very confusing.  
-  Review the subtyping results

#### Is there anything that you want to discuss further?
Observed 5 `DMG, H3 K28`  samples that can be subtyped as `IHG, ROS1-altered`. Seems not mutually exclusive. 

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
YES 

#### What types of results are included (e.g., table, figure)?
v12 HGG subtyping results  

#### What is your summary of the results?
v12 HGG subtyping results 

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by jharenza** [comment]
I think I got all of the logic updated here, but now trying to remedy some missing files in the final output tables created in script 07, but calling it quits for tonight.

- Many targeted panel RNA fusion are missing due to [this comment ](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/321#discussion_r1125591958) - missing in fusion summary
- A handful of DNA are missing - I think due to missing in MAF: "BS_DBH8YWA1" "BS_EE73VE7V" "BS_XNYQS1WG" "BS_XQF18WZP"
- Some RNA-Seq missing, have to check this still
- A handful of methyl missing, have to check this as well

File: analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd [hunk 1]
```diff
@@ -55,35 +61,62 @@ if (!dir.exists(results_dir)) {
...
+# Samples included on the basis of the pathology diagnosis fields
+hgg_only_meta <- metadata %>%
+  # Inclusion on the basis of CBTTC harmonized pathology diagnoses
+  filter(pathology_diagnosis %in% path_dx_list$exact_path_dx |
+         # Inclusion based on pathology free text diagnosis
+         pathology_free_text_diagnosis == path_dx_list$gliomatosis_path_free_text_exact |
+         # Inclusion based on pathology free text diagnosis for IHG
+         pathology_free_text_diagnosis  %in% path_dx_list$IHG_path_free_path_dx)
+
+dna_bsids <- hgg_meta %>%
```
Comment (zzgeng) [inline analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd]: `hgg_meta` is not created in previous code, Do you mean `hgg_only_meta`?

Comment (jharenza) [reply analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd]: good catch!

**Comment by jharenza** [comment]
@rjcorb and @zzgeng this is ready for review. Specifically, please pay attention to script 07 which does the subtyping. The rules are:
1. If K28M present or in the absence of mutation or no mutation data, methyl K28M subtype --> DMG, H3 K28
2. If G35R/V present or methyl G35 subtype if no mutation data --> DHG, H3 G35
3. If infant type hemispheric via free text or IHG by methyl and not already subtyped as above, then use fusions to determine which IHG class (eg ALK, ROS, NTRK) --> IHG, xxxx or if no fusion detected, but is IHG by free text or methyl --> IHG, To be classified
4. If high-confidence methylation was H3, wildtype and no other subtypes above were determined --> HGG, H3 wildtype
5. All others --> HGG, To be classified

Note: I created a unique mapping `id` - for tumors, it is simply a concat of sample_id and composition. For cell lines, it is a concat of sample_id, composition, cell_line_composition, and cell_line_passage. This enables subtype propagation if called via one biospecimen (for eg methyl).

The final table is one row per bs_id and should match the hgg_subset number of bs_ids. The final lesion+subtype table is one row with each combination of DNA, RNA, Methyl. I do not prefer this, but the module was set up this way in the past and as such, many bs_ids are repeated such to create the combinations, fyi.

Summary of tumors subtyped:
```r
Var1	Freq
DMG, H3 K28	278			
HGG, H3 wildtype	256			
HGG, To be classified	87			
DHG, H3 G35	35			
HGG, IDH	15			
IHG, NTRK-altered	4			
IHG, ROS1-altered	4			
IHG, ALK-altered	3			
IHG, To be classified	1			
```

**Comment by zzgeng** [comment]
136 samples in `HGG_molecular_subtype.tsv` are missing `Kids_First_Participant_ID` and `sample_id`, which are all the `HGG, To be classified` samples. For example: 

<!DOCTYPE html>

NA | NA | BS_05EMADFS | 7316-287_Solid Tissue | HGG, To be classified
-- | -- | -- | -- | --
NA | NA | BS_05QV0A71 | 7316-6251_Solid Tissue | HGG, To be classified

File: analyses/molecular-subtyping-HGG/07-HGG-molecular-subtyping-combine-table.Rmd [hunk 2]
```diff
@@ -631,75 +406,129 @@ MET <- fusion_df %>%
...
+# which ids have samples which can be assigned subtypes
+all_subtypes <- subtype_map %>%
+  left_join(hgg_meta[,c("Kids_First_Biospecimen_ID", "id")]) %>%
+  select(Kids_First_Biospecimen_ID, id, molecular_subtype)
+
+# add rest as TO BE CLASSIFIED
+to_be_classified <- hgg_meta %>%
+  filter(!id %in% all_subtypes$id) %>%
+  mutate(molecular_subtype = "HGG, To be classified") %>%
+  select(Kids_First_Biospecimen_ID, id, molecular_subtype)
```
Comment (zzgeng) [inline analyses/molecular-subtyping-HGG/07-HGG-molecular-subtyping-combine-table.Rmd]:
```
select(Kids_First_Biospecimen_ID, id, molecular_subtype)
```
change to 

```
select(Kids_First_Biospecimen_ID, id, molecular_subtype, Kids_First_Participant_ID, sample_id)
```

Comment (jharenza) [reply analyses/molecular-subtyping-HGG/07-HGG-molecular-subtyping-combine-table.Rmd]: updated in [87a9ed4](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/331/commits/87a9ed4fc97315ff00995058767888276bdb16db)

**Review by zzgeng — APPROVED** [review]
Code works well and logic is sound!

**Review by rjcorb — APPROVED** [review]
LGTM! Code runs fine and can confirm all hgg tumor bs_ids are in final table.

**Pull request merged**